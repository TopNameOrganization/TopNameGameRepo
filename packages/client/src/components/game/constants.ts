export const TileSize = 32;
export enum Tile {
  Empty = 0,
  Brick = 1,
  Concrete = 2,
  Stair = 3,
  Rope = 4,
  Bonus = 5,
  Enemy = 6,
  Player = 7,
  Out = 100,
}

export const VELOCITY = 200;
export enum Orientation {
  Left = 0,
  Right = 1,
}
export enum RunnerAction {
  Stay = 0,
  Fall = 1000,
  MoveLeft = 37,
  MoveUp = 38,
  MoveRight = 39,
  MoveDown = 40,
}

export enum AnimationPhase {
  Stay = 0,
  Run = 1,
  Climb = 2,
  Hang = 3,
  Fall = 4,
}
