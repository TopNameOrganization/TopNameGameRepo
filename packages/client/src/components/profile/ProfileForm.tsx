import * as React from 'react';
import {useFormik} from 'formik';
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';
import Box from '@mui/material/Box';
import {validationProfile} from "./validation";

interface IProfile {
    [key: string]: FormDataEntryValue | null
}

export default function ProfileForm() {
    const [isDisabled, setIsDisabled] = React.useState<boolean>(false);

    const formik = useFormik({
        initialValues: {
            firstName: '',
            secondName: '',
            nickName: '',
            phone: '',
            email: '',
            password: '',
        },
        validationSchema: validationProfile,
        onSubmit: (profileData: IProfile) => {
            console.log('profileData: ', profileData)
            setIsDisabled(false);
        },
    });

    const handleChangeIsDisabled = () => {
        setIsDisabled(true);
    };

    return (
        <Box component="form" onSubmit={formik.handleSubmit} noValidate sx={{ mt: 1 }}>
            <TextField
                variant="standard"
                margin="dense"
                fullWidth
                name="firstName"
                label="First name"
                type="text"
                id="firstName"
                autoComplete="firstName"
                disabled={!isDisabled}
                onChange={formik.handleChange}
                error={formik.touched.firstName && Boolean(formik.errors.firstName)}
                helperText={formik.touched.firstName && formik.errors.firstName}
            />
            <TextField
                variant="standard"
                margin="dense"
                fullWidth
                name="secondName"
                label="Second name"
                type="text"
                id="secondName"
                autoComplete="secondName"
                disabled={!isDisabled}
                onChange={formik.handleChange}
                error={formik.touched.secondName && Boolean(formik.errors.secondName)}
                helperText={formik.touched.secondName && formik.errors.secondName}
            />
            <TextField
                variant="standard"
                margin="dense"
                required
                fullWidth
                name="nickName"
                label="Nick name"
                type="text"
                id="nickName"
                autoComplete="nickName"
                disabled={!isDisabled}
                onChange={formik.handleChange}
                error={formik.touched.nickName && Boolean(formik.errors.nickName)}
                helperText={formik.touched.nickName && formik.errors.nickName}
            />
            <TextField
                variant="standard"
                margin="dense"
                required
                fullWidth
                id="email"
                label="Email Address"
                name="email"
                autoComplete="email"
                disabled={!isDisabled}
                value={formik.values.email}
                onChange={formik.handleChange}
                error={formik.touched.email && Boolean(formik.errors.email)}
                helperText={formik.touched.email && formik.errors.email}
            />
            <TextField
                variant="standard"
                margin="dense"
                fullWidth
                name="phone"
                label="Phone"
                type="tel"
                id="phone"
                autoComplete="phone"
                disabled={!isDisabled}
                onChange={formik.handleChange}
                error={formik.touched.phone && Boolean(formik.errors.phone)}
                helperText={formik.touched.phone && formik.errors.phone}
            />
            <TextField
                variant="standard"
                margin="dense"
                required
                fullWidth
                name="password"
                label="Password"
                type="password"
                id="password"
                autoComplete="current-password"
                disabled={!isDisabled}
                value={formik.values.password}
                onChange={formik.handleChange}
                error={formik.touched.password && Boolean(formik.errors.password)}
                helperText={formik.touched.password && formik.errors.password}
            />
            {
                isDisabled
                    ? (<Button
                        type="submit"
                        fullWidth
                        variant="contained"
                        sx={{ mt: 5, mb: 2 }}
                    >
                        Save Profile
                    </Button>)
                    : (<Button
                        onClick={handleChangeIsDisabled}
                        type="button"
                        fullWidth
                        variant="contained"
                        sx={{ mt: 5, mb: 2 }}
                    >
                        Edit Profile
                    </Button>)
            }
        </Box>
    );
}
