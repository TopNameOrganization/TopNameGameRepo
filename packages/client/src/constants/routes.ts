export const ROUTES = {
  root: '/',
  home: '/home',
  game: '/game',
  login: '/login',
  signup: '/signup',
  profile: '/profile',
  leaderBoard: '/leader-board',
  forum: '/forum',
  forumCreate: '/forum/create',
  forumDetail: '/forum/:id',
}
