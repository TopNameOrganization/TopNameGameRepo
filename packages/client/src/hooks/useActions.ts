import {useDispatch} from "react-redux";
import {bindActionCreators} from "@reduxjs/toolkit";
import Actions from '../store/actions/index'

export const useActions = () => {
  const dispatch = useDispatch()
  return bindActionCreators(Actions, dispatch)
}
