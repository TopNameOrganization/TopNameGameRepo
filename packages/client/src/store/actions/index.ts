import * as UserAction from './userAction'
import * as UserProfileAction from './changeUserProfileAction'

export default {
  ...UserAction,
  ...UserProfileAction
}
