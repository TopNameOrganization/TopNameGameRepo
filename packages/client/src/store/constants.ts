import {UserState} from "./types";

export const initialState: UserState = {
  user: {
    id: null,
    first_name: '',
    second_name: '',
    display_name: '',
    login: '',
    email: '',
    phone: '',
    avatar: ''
  },
  loading: false,
  error: null
}
