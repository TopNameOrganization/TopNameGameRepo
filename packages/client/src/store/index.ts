import { rootReducer, RootState } from './reducers'
import { configureStore } from "@reduxjs/toolkit";

export const setupStore = (initialState?: RootState) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState: initialState
  });
}
