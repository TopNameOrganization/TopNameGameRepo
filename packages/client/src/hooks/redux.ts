import {TypedUseSelectorHook, useDispatch, useSelector} from "react-redux";
import {AppDispatch, RootState} from "../store/reducers";
import {bindActionCreators} from "@reduxjs/toolkit";
import Actions from "../store/actions";

export const useAppDispatch = () => useDispatch<AppDispatch>()
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
export const useAppActions = () => bindActionCreators(Actions, useAppDispatch())
