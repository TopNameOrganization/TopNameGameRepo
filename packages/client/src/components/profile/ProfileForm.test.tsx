import * as renderer from 'react-test-renderer';
import { ProfileForm } from './ProfileForm';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import * as appHooks from "../../hooks/redux";
import { Provider } from "react-redux";
import { userInitialState } from "../../store/constants";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const useAppSelectorMock = jest.spyOn(appHooks, 'useAppSelector');

describe('Profile form', () => {
  const useForm = true;
  const showForm = false;

  it('should renders profile form correctly', () => {
    useAppSelectorMock.mockReturnValue(userInitialState)
    const store = mockStore(userInitialState)
    const tree = renderer
        .create(
            <Provider store={store}>
                <ProfileForm noUseForm={useForm} isDisabled={showForm} />
            </Provider>)
        .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
