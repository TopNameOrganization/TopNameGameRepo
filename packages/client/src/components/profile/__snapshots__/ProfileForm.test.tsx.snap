// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile form should renders profile form correctly 1`] = `
<form
  className="MuiBox-root css-164r41r"
  id="profileForm"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-disabled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="first_name"
      id="first_name-label"
    >
      First name
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="first_name"
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input Mui-disabled css-1x51dt5-MuiInputBase-input-MuiInput-input"
        disabled={true}
        id="first_name"
        name="first_name"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-disabled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="second_name"
      id="second_name-label"
    >
      Second name
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="second_name"
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input Mui-disabled css-1x51dt5-MuiInputBase-input-MuiInput-input"
        disabled={true}
        id="second_name"
        name="second_name"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-disabled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="display_name"
      id="display_name-label"
    >
      Nick name
      <span
        aria-hidden={true}
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="display_name"
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input Mui-disabled css-1x51dt5-MuiInputBase-input-MuiInput-input"
        disabled={true}
        id="display_name"
        name="display_name"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-disabled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="login"
      id="login-label"
    >
      login
      <span
        aria-hidden={true}
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="login"
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input Mui-disabled css-1x51dt5-MuiInputBase-input-MuiInput-input"
        disabled={true}
        id="login"
        name="login"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-disabled Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="email"
      id="email-label"
    >
      Email Address
      <span
        aria-hidden={true}
        className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         
        *
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="email"
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input Mui-disabled css-1x51dt5-MuiInputBase-input-MuiInput-input"
        disabled={true}
        id="email"
        name="email"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1z10yd4-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-disabled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="phone"
      id="phone-label"
    >
      Phone
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="phone"
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input Mui-disabled css-1x51dt5-MuiInputBase-input-MuiInput-input"
        disabled={true}
        id="phone"
        name="phone"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="tel"
        value=""
      />
    </div>
  </div>
</form>
`;
